### ISSUE TITLE
[audio] Not all types of WAV files supported

### ISSUE THREAD

**Issue by Obscure2020** [raysan5/raylib#1340]
- [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [x] My code has no errors or misuse of raylib

### Issue description

I have been testing out raylib's music streaming abilities by changing `audio_music_stream.c` to refer to other sound files and compiling it to see how well it works. In my testing I have come across something strange: while raylib _does_ support streaming WAV files, no audio will actually be played unless the WAV file used 16-bit samples.

I tested `audio_music_stream.c` with 16-bit, 24-bit, and 32-bit WAV files. Only the 16-bit ones actually played sound. However, the 24-bit and 32-bit WAVs would not cause compilation or runtime errors, and in fact, the music-playing progress bar featured in `audio_music_stream.c` still functions perfectly.

[Here is a Google Drive download link](https://docs.google.com/uc?export=download&id=1m_wd8rGSXX2Zocx0V7OcwC7HA2Qk5JLt) for a ZIP folder containing all the WAV files I used to test this.
I would have just attached the ZIP folder, but it was too big. (WAVs are large.)

**Comment by raysan5** [comment]
It should work now. Feel free to reopen if it still fails.

**Comment by istvanfedak** [comment]
@raysan5 I've been experiencing the same issue with the kitten.wav sound:
- [test-sound.zip](https://github.com/raysan5/raylib/files/8767475/test-sound.zip)

When you run the program the wav file won't be played but I can play it with an audio player on my computer. I did some more digging and the kitten.wav file seems to be a newer .wav format

**Comment by Obscure2020** [comment]
> I did some more digging and the kitten.wav file seems to be a newer .wav format

I'm not certain it's "newer."
The file is more akin to "wrong."

I ran your `kitten.wav` through `ffprobe` to see what exactly was inside. Result:
```
Input #0, wav, from 'kitten.wav':
  Duration: 00:00:01.69, bitrate: 56 kb/s
  Stream #0:0: Audio: mp3 (U[0][0][0] / 0x0055), 16000 Hz, stereo, fltp, 56 kb/s
```

I had to look this up, because I haven't seen this sort of thing in years.
Turns out it is valid to put an mp3 stream in a `.wav` file. The `wav` format does provide for that.
However, that's NOT AT ALL a common or recommended practice nowadays.
It's much more common to find mp3 audio in an actual `.mp3` file.

Also, I doubt this particular file is "new" by any stretch of the imagination.
`ffprobe` says it's a 16000Hz stream encoded at 56kb/s.
I mean, seriously? How cruddy can you get?? Modern standards are at _least_ 44100Hz and usually at least 128kb/s.
I'm guessing this is an ANCIENT file.

So long story short, I don't think this is an edge case raylib needs to account for.
Your `.wav` isn't really a wav.

Oh, and for free, I'm giving you a gift: [kitten.zip](https://github.com/raysan5/raylib/files/8767538/kitten.zip), which contains `kitten.mp3`.
I used `ffmpeg` to copy the mp3 bitstream from `kitten.wav` into a new `kitten.mp3` file for you.
It's the exact same audio, not converted, not transcoded.
I just put it into the container it should have been in to begin with. Enjoy.
